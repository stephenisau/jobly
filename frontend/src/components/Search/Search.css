input {
    width: 100%;
    border: 80%;
}