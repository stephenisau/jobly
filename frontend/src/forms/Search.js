import React, { Component } from "react";
import { debounce } from "lodash";

class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      search: ""
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange = (evt) => {
    this.setState({ 
      [evt.target.name]: evt.target.value 
    });
  };

  handleSubmit = debounce((evt) => {
    evt.preventDefault()
    const { search } = this.state;
    this.props.search(search);
  }, 500);

  render() {
    return (
      <div className="md-form mt-0">
        <form onChange={this.handleSubmit}>
          <input name="search" 
            id="searchbar"
            value={this.state.search} 
            onChange={this.handleChange} 
            placeholder="Search" 
            aria-label="Search"
            type="search"/>
          {/* <button className="btn btn-primary" type="submit">Submit</button> */}
        </form>
      </div>
    );
  }
}

export default Search;