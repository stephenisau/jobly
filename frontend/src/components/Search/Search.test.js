import React from "react";
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import Search from "../Forms/Search";

it('renders without crashing', function() {
  shallow(<Search/>);
});

it('matches snapshot', function() {
  let wrapper = shallow(<Search/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it ('state reflects change in input', function() {
  const wrapper = mount(<Search />);
  let input = wrapper.find('#searchbar');

  input.instance().value = 'search';
  input.simulate('change');
  
  expect(wrapper.state('search')).toEqual('search');
})


