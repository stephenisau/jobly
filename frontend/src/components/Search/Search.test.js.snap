// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="md-form mt-2"
>
  <form
    onChange={[Function]}
  >
    <input
      aria-label="Search"
      id="searchbar"
      name="search"
      onChange={[Function]}
      placeholder="Search Jobly here..."
      type="search"
      value=""
    />
  </form>
</div>
`;
